app_flutter/lib/status_grid.dart

```diff
@@ -99,64 +99,152 @@ class StatusGrid extends StatelessWidget {
-                /// [Task] was skipped so don't show anything.
-                return const SizedBox();
-              }
-
-              return TaskBox(
-                task: task,
-                buildState: buildState,
-              );
-            },
+      child: ListView(
+        children: rows,
+        shrinkWrap: false,
+      ),
+    );
+  }
+
+  List<Widget> _buildRows() {
+    final List<Widget> rows = <Widget>[];
+
+    rows.add(
+      Container(
+        height: 50,
+        child: NotificationListener<ScrollNotification>(
+          child: ListView(
+            controller: controllers[0],
+            children: _buildTaskIconRow(),
+            scrollDirection: Axis.horizontal,
           ),
+          onNotification: (ScrollNotification scrollInfo) {
+            _syncScroller.processNotification(scrollInfo, controllers[0]);
+            return;
+          },
         ),
       ),
     );
+
+    for (int rowIndex = 0; rowIndex < widget.taskMatrix.rows; rowIndex++) {
+      final List<TaskBox> tasks = <TaskBox>[];
+      for (int colIndex = 0; colIndex < widget.taskMatrix.columns; colIndex++) {
+        tasks.add(TaskBox(
+          buildState: widget.buildState,
+          task: widget.taskMatrix.task(rowIndex, colIndex),
+        ));
+      }
+
+      rows.add(
+        Container(
+          height: 50,
+          child: NotificationListener<ScrollNotification>(
+            child: ListView.builder(
+              controller: controllers[rowIndex + 1],
+              scrollDirection: Axis.horizontal,
+              itemCount: widget.taskMatrix.columns + 1,
+              itemBuilder: (BuildContext context, int colIndex) {
+                if (colIndex == 0) {
+                  return CommitBox(
+                    commit: widget.statuses[rowIndex].commit,
+                  );
+                }
+                return Container(
+                  width: 50,
+                  child: TaskBox(
+                    buildState: widget.buildState,
+                    task: widget.taskMatrix.task(rowIndex, colIndex - 1),
+                  ),
+                );
+              },
+              shrinkWrap: false,
+            ),
+            onNotification: (ScrollNotification scrollInfo) {
+              _syncScroller.processNotification(
+                  scrollInfo, controllers[rowIndex + 1]);
+              return;
+            },
+          ),
+        ),
+      );
+    }
+
+    return rows;
+  }
+
+  List<Widget> _buildTaskIconRow() {
+    final List<Widget> taskIcons = <Widget>[];
+    taskIcons.add(Container(width: 50));
+    for (int colIndex = 0; colIndex < widget.taskMatrix.columns; colIndex++) {
+      taskIcons.add(
+        Container(
+          width: 50,
+          child: TaskIcon(
+            task: widget.taskMatrix.sampleTask(colIndex),
+          ),
+        ),
+      );
+    }
+
+    return taskIcons;
+  }
+}
+
+// https://stackoverflow.com/questions/54859779/scroll-multiple-scrollable-widgets-in-sync
```

Comment:
Please add a doc comment explaining what this does. The link is good for context, but a one sentence summary is nice

Reply:
Good catch! Added.

Just left over from hacking together the experiment PR.